usr/share/lib/img_proof/tests/SLES/test_sles_grow_root.py

```diff
@@ -0,0 +1,44 @@
+import shlex
+
+
+def test_sles_grow_root(host):
+    """
+    Ensure root filesystem grows to non default size
+    """
+
+    # Get root disk size
+    root_part = host.run('findmnt -n -f -o SOURCE /').stdout.strip()
+
+    device_name = host.run(
+        'lsblk -npo pkname {part}'.format(part=root_part)
+    ).stdout.strip()
+
+    result = host.run(
+        'lsblk -ndo size {device}'.format(device=device_name)
+    ).stdout.strip()
+
+    disk_size = int(result.replace('G', ''))
+
+    # Get root partition size
+    result = host.run(
+        'df -BG {part} | sed 1D'.format(part=root_part)
+    ).stdout.strip()
+
+    root_size = shlex.split(result)[1]  # Filesystem 1G-blocks
+    root_size = int(root_size.replace('G', ''))
+
+    # Get boot partition size
+    boot_part = host.run('findmnt -n -f -o SOURCE /boot').stdout.strip()
```

Comment:
Sorry I didn't provide all the details, there is a wrinkle which currently falls into the rounding error category. We should at least provide a comment. But it could be coded with an if condition.

For images where we build for UEFI/EFI, GCE ATM, kiwi automatically creates an efi partition that gets mounted to `/boot/efi` and as such `findmnt -n -f -o SOURCE /boot` will return nothing because the mount point is `/boot/efi`. If `/boot/efi` grows sufficiently large then we end up with

total_size = / + /boot + /boot/efi

however, since at present the efi partition is < 1 GB the rounding error in the output of `df -h` hides this. I don't think /boot/efi will grow to >= 1GB as such it is possibly not worth the effort to do the additional parsing of the output and looking for an efi partition. But a comment in the code to this effect is definitely helpful to help us in the future should the efi partition grow unexpectedly large.

Reply:
I noticed this, the problem is that it becomes a slippery slope. Azure images have I think 4 partitions. If we are adding conditionals for efi then we probably need conditionals for the other Azure partitions. IMHO it's not worth chasing all of these and only update things if necessary in the future.

Reply:
I will add a comment to cover this...